scripts/publish.js

```diff
@@ -0,0 +1,161 @@
+/* eslint-disable camelcase */
+import inquirer from "inquirer";
+import { invalid, section, valid } from "../utils.js";
+import ora from "ora";
+import { apiClient } from "./utils/apiClient.js";
+
+const POLL_INTERVAL = 2000;
+
+const IN_PROGRESS_STATUSES = ["PENDING", "STARTED"];
+
+const waitForTaskToComplete = async (taskId) => {
+  const waitingSpinner = ora(
+    "Waiting for publish to complete. This may take a few minutes.\n\n Publishing will continue if you exit here (CTRL+C)."
+  ).start();
+
+  let latestProcessingResultBody;
+
+  // Poll indefinitely until the task is resolved.
+  while (true) {
+    await new Promise((resolve) => {
+      setTimeout(() => {
+        resolve();
+      }, POLL_INTERVAL);
+    });
+
+    const processingResult = await apiClient.get({
+      path: "/v2/status-of-published-module/" + taskId
+    });
+
+    if (!processingResult.ok) {
+      waitingSpinner.stop();
```

Comment:
It seems like the waiting spinner will stop at the end after you break the loop at line 45 and it reads to line 64. Might  not need this